validator/src/main.rs

```diff
@@ -183,28 +182,73 @@ fn initialize_ledger_path(
     )
     .map_err(|err| err.to_string())?;
 
-    let rpc_addr = nodes
-        .iter()
-        .filter_map(|contact_info| {
-            if contact_info.gossip == entrypoint.gossip
-                && ContactInfo::is_valid_address(&contact_info.rpc)
-            {
-                Some(contact_info.rpc)
-            } else {
-                None
-            }
-        })
-        .next()
-        .unwrap_or_else(|| {
-            error!(
-                "Entrypoint ({:?}) is not running the RPC service",
-                entrypoint.gossip
-            );
-            exit(1);
-        });
+    let rpc_addr = nodes.iter().find_map(|contact_info| {
+        if contact_info.gossip == entrypoint.gossip
+            && ContactInfo::is_valid_address(&contact_info.rpc)
+        {
+            Some(contact_info.rpc)
+        } else {
+            None
+        }
+    });
+
+    if let Some(rpc_addr) = rpc_addr {
+        Ok((rpc_addr, RpcClient::new_socket(rpc_addr)))
+    } else {
+        Err(format!(
+            "Entrypoint ({:?}) is not running the RPC service",
+            entrypoint.gossip
+        ))
+    }
+}
 
-    let client = RpcClient::new_socket(rpc_addr);
-    let genesis_hash = client.get_genesis_hash().map_err(|err| err.to_string())?;
+fn check_vote_account(
+    rpc_client: &RpcClient,
+    vote_pubkey: &Pubkey,
+    voting_pubkey: &Pubkey,
+    node_pubkey: &Pubkey,
+) -> Result<(), String> {
+    let found_account = rpc_client
```

Comment:
@mvines Come to think of this richness of sanity checks, I think checking minimum lamports in `node_pubkey` might also be helpful. Should add that? Is there any concern?

Reply:
Yes that'd be great.  I don't know what a minimum should be right now so let's just check for non-zero as a start

Reply:
I ended up checking existince of node account and account balance being at least `1 lamport`. After all there should be no account with `0 lamports`. Although, I had [this bug](#6890) in my mind.